barretenberg/cpp/src/barretenberg/stdlib/primitives/bool/bool.cpp

```diff
@@ -458,120 +436,72 @@ bool_t<Builder> bool_t<Builder>::conditional_assign(const bool_t<Builder>& predi
     }
 
     bool same = lhs.witness_index == rhs.witness_index;
-    bool witness_same = same && lhs.witness_index != IS_CONSTANT && (lhs.witness_inverted == rhs.witness_inverted);
-    bool const_same = same && (lhs.witness_index == IS_CONSTANT) && (lhs.witness_bool == rhs.witness_bool);
+    bool witness_same = same && !lhs.is_constant() && (lhs.witness_inverted == rhs.witness_inverted);
+    bool const_same = same && lhs.is_constant() && (lhs.witness_bool == rhs.witness_bool);
     if (witness_same || const_same) {
         return lhs;
     }
     return (predicate && lhs) || (!predicate && rhs);
 }
 
+/**
+ * @brief Implements implication operator in circuit.
+ */
 template <typename Builder> bool_t<Builder> bool_t<Builder>::implies(const bool_t<Builder>& other) const
```

Comment:
`imply`, `must_imply` and `implies_both_ways` were added for aztec connect if I remember correctly. If these are not being used anywhere (including noir), I am in favour of removing them. Ideally, I think bool class should only contain standard operations (`!, |, ^, &`).

Reply:
I think only `must_imply` is currently being used in ecdsa verifier and some places in recursive verifier.